miscellaneous/csv.md

```diff
@@ -0,0 +1,195 @@
+    column("first_name", user.first_name)
+    column("last_name", user.last_name)
+    column("email", user.email)
+  end
+end
+```
+
+And we can call it the usual way:
+
+::: code-group
+```ruby
+users = User.all
+
+UserExport.new(users).call
+```
+```csv
+id,first_name,last_name,email
+1,Buster,Bluth,buster@example.com
+2,Maeby,Fünke,maeby@example.com
+```
+:::
+
+Nice! Ok, so what's actually happening? We initialize the CSV component with a collection of users, and when we call `call` on the component, it executes our `row_template` method for each item in the collection. The `column` method is pushing the header name, and the corresponding value into the CSV.
+
+## Safety
+
+There are some really cool features I want to show you about `Phlex::CSV`, but first we need to talk about safe CSVs. Spreadsheet programs that are often used to view CSVs are vulnerable to injection attacks when cells are interpreted as containing formulae. However CSVs are often used as a data interchange format between computer programs, and "escaping" values for a spreadsheet program would just be mangling your data.
+
+So there's unfortunately no good default assumption we can make. So you have to make it. If the intended use for your CSVs are to be consumed by humans, likely through a spreadsheet program, then you need to define:
+
+```ruby
+def escape_csv_injection? = true
+```
+
+If, on the other hand, you're producing CSVs that are meant to be imported by some other software, and you need the data to be unaltered, you should define:
+
+```ruby
+def escape_csv_injection? = false
+```
+
+If you do not set this method, you will get a runtime error raised when you run `call`.
+
+For more information about CSV formula injection, see [OWASPs explanation](https://owasp.org/www-community/attacks/CSV_Injection) on it.
+
+If, like many people, almost all of your CSV reports fall in to one category or another, it can be easiest to define a base class in your application with this method defined:
+
+```ruby
+class ApplicationCSV < Phlex::CSV
+  def escape_csv_injection? = true
+end
+```
+
+## Trimming whitespace
+
+Depending on the source of your data, it can be helpful to have the values trimmed of any leading or trailing whitespace. To opt-in to this, you can define a `trim_whitespace?` method. By default, it is `false`, and preserves any value you pass to a `column`.
+
+```ruby
+class ApplicationCSV < Phlex::CSV
+  def trim_whitespace? = true
+end
+```
+
+With this method defined, values will have `strip` called on them before they are inserted into the resulting CSV.
+
+## Skipping headers
+
+And what if we don't care about CSV headers? Just override `render_headers?` to false. You can keep or remove the header argument passed to `column`. If it's helpful for readability, you can keep it and it will be ignored. Or if it's clear enough without it, toss them.
+
+::: code-group
+```ruby
+class UserExport < ApplicationCSV
+  def render_headers? = false
```

Comment:
Perhaps here or in the section below about the initializer, we should highlight that you could pass this in as an option if you wanted define a method that points to an ivar.